lib/exception_notifier/slack_notifier.rb

```diff
@@ -1,27 +1,43 @@
 module ExceptionNotifier
   class SlackNotifier
-
-    attr_accessor :notifier
+    attr_accessor :slack_options
 
     def initialize(options)
-      begin
-        webhook_url = options.fetch(:webhook_url)
-        @message_opts = options.fetch(:additional_parameters, {})
-        @notifier = Slack::Notifier.new webhook_url, options
-      rescue
-        @notifier = nil
-      end
+      self.slack_options = options
     end
 
     def call(exception, options={})
-      message = "An exception occurred: '#{exception.message}' on '#{exception.backtrace.first}'"
-      @notifier.ping(message, @message_opts) if valid?
-    end
+      env = options[:env]
 
-    protected
+      link = env['HTTP_HOST'] + env['REQUEST_URI']
+      title = "#{env['REQUEST_METHOD']} <http://#{link}|http://#{link}>\n"
+
+      message = "------------------------------------------------------------------------------------------\n"
+      message += "*Project:* #{Rails.application.class.parent_name}\n"
+      message += "*Environment:* #{Rails.env}\n"
+      message += "*Time:* #{Time.zone.now.strftime('%Y-%m-%d %H:%M:%S')}\n"
+      message += "*Exception:* `#{exception.message}`\n"
+
+      req = Rack::Request.new(env)
+      unless req.params.empty?
+        message += "*Parameters:*\n"
+        message += req.params.map { |k, v| ">#{k}=#{v}" }.join("\n")
```

Comment:
It seems appropriate to only include `filtered_parameters` in the slack message to avoid leaking any sensitive information.
https://github.com/rails/rails/blob/fe8a6ba3a12452c9ba9cd4da554e65b08df2155c/actionpack/lib/action_dispatch/http/filter_parameters.rb#L35-L38